src/systems/filecoin_blockchain/struct/block/block.id

```diff
@@ -45,12 +45,41 @@ type BlockHeader struct {
 
 // TODO: remove this. header is signed
 type SignedBlock struct {
-    block  &Block
-    sig    filcrypto.Signature
+    BlockCID   &Block
+    MinerAddr  addr.Address
+    Signature  filcrypto.Signature
 }
 
 type Block struct {
     Header    BlockHeader
     Messages  [msg.Message]
     Receipts  [msg.MessageReceipt]
 }
+
+// Checkpoint represents a particular block to use as a trust anchor
+// in Consensus and ChainSync
+//
+// Note: a Block uniquely identifies a tipset (the parents)
+// from here, we may consider many tipsets that _include_ Block
+// but we must indeed include t and not consider tipsets that
+// fork from Block.Parents, but do not include Block.
+type Checkpoint BlockCID
```

Comment:
With tipsets, using a single block here confuses me a bit. Things that would make more transparent sense would be a state CID, or a tipset identifier (list of block CIDs).

Reply:
So how about a state CID instead of a Block CID?

Note that the genesis state is not constructable as a sequence of messages applied to an empty state (it contains system actors already installed). We actually don't need a genesis block so much as a genesis state. What would break if multiple miners produced blocks in round 1, all sharing the same parent state root?

Reply:
> So how about a state CID instead of a Block CID? 

I'm not seeing how this is better? maybe im missing it? 

I understand some of the motivation -- the genesis block seems superfluous, in light that it points back to a parent `StateTree` anyway. And I see that the algorithms here can be changed to reason also about `StateTrees` as checkpoints, and to deal with cases of syncing blocks back to a `StateTree` instead of syncing blocks back to another block (this would add a lot `StateTree` oriented checks in addition to all the block oriented checks we already need).

this would change the logic of tools and processes/expectations around checkpoints that already exist in broader community w/ other chains -- this can be changed, but we would have to make sure a lot of tools supported looking up `StateTree` by cid (tools which already _would_ support lookup of blocks by cid but not necessarily a `StateTree`, eg simple block explorers, wallets, chain exporters, etc). this is different and would need more work, but doable. 

Im not seeing lots of benefit for these changes and additional tooling work -- but maybe im missing them? is there more beyond the conceptual re-writing and cleaning up of checkpoints? (namely, that they always point to a "thing" at a particular round of which there is only one valid instance-- ie only one `StateTree` per round)

I would be OK doing this (changing to use `StateTree` as the checkpoint) if `StateTree` also became the "root object" and included the blocks within it (as in ETH2, and some other proposed chains, which treat consensus as a VM-internal process, not VM-surrounding process). Making `StateTree` the root object would cement its standing as a key, root data structure, that would need to get support in a ton of tools. (but would still have a genesis block, see below.)

> Note that the genesis state is not constructable as a sequence of messages applied to an empty state (it contains system actors already installed). 

Yes

> We actually don't need a genesis block so much as a genesis state. 

I believe we do need a genesis block -- we derive randomness from the ticket there. Rather than special casing, it is easier/less complex to ensure a well-formed chain always, including at the beginning

> What would break if multiple miners produced blocks in round 1, all sharing the same parent state root?

invariants/tools expecting the chain to end at one block (this reminds of git's empty repo and the edge cases it introduces, instead of starting always with the same empty/first commit). 

In any case, the mainnet filecoin chain will need to start with a small chain stub of blocks.
- product reason, we actually have to include some data in different blocks. (dont worry about this at all till Q1 -- we'll produce the relevant chain stub)
- A lot of code expects lookbacks, especially actor code.  Rather than introducing a bunch of special case logic for what happens ostensibly once in network history (special case logic which adds complexity and likelihood of problems), it is easiest to assume the chain is always at least X blocks long, and the system lookback parameters are all fine and dont need to be scaled in the beginning of network's history.

Reply:
Ok, thanks. You understand my motivations and this proposal is reasonable.